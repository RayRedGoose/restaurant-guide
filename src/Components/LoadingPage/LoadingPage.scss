@import "../../assets/css/variables.scss";

.loading {
  width: 100%; height: 60vh;
  display: flex;
  align-items: center;
  justify-content: center;
  img {
    animation: rotate 6s cubic-bezier(0.250, 0.460, 0.450, 0.940) infinite;
    width: 4rem;
  }
  h2 {
    color: $accent-color;
    font-family: $accent-font;
    font-size: 2rem;
    font-weight: 700;
    letter-spacing: 0.05em;
    margin-left: 1rem;
  }
}

@keyframes rotate {
  0% {
    transform: rotate(0deg);
  }
  25% {
    transform: rotate(25deg);
  }
  75%{
    transform: rotate(-25deg);
  }
  100% {
    transform: rotate(0deg);
  }
}
