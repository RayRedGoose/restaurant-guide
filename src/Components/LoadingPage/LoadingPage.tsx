import "./LoadingPage.scss";
import React from "react";
import loading from "assets/images/loading.svg";

const LoadingPage: React.FC = () => (
  <section className="loading">
    <img src={loading} alt="loading" />
    <h2>Loading...</h2>
  </section>
);

export default LoadingPage;
