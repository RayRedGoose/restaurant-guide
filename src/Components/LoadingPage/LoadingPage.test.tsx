import React from 'react';
import { shallow } from 'enzyme';
import LoadingPage from './LoadingPage';

it('should match the snapshot', () => {
  const wrapper = shallow(<LoadingPage />);
  expect(wrapper).toMatchSnapshot();
});
