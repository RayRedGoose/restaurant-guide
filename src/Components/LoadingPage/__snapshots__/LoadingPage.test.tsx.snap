// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<section
  className="loading"
>
  <img
    alt="loading"
    src="loading.svg"
  />
  <h2>
    Loading...
  </h2>
</section>
`;
